Moved CSS location to media folder

ashsmith opened the pull request:
Fixes #5

Commit:
Moved CSS location to media folder

Fixes #5

bobbyshaw commented:
+1

ashsmith commented on src/app/code/community/Meanbee/ConfigPoweredCss/Model/Config.php:
```diff
@@ -77,6 +78,66 @@ public function getCssFilename($store = null)
+    public function getCssDirectoryPath($store = null)
+    {
+
+        return Mage::getBaseDir('media') . $this->getCssPath($store);
```
Needs a trailing slash after media directory `Mage::getBaseDir('media') . DS . $this->getCssPath($store);`

ashsmith commented on src/app/code/community/Meanbee/ConfigPoweredCss/Model/Config.php:
```diff
@@ -77,6 +78,66 @@ public function getCssFilename($store = null)
+     */
+    public function getCssFileUrl($store = null)
+    {
+        return Mage::getUrl(sprintf('media/%s', $this->getCssPath($store))) . $this->getCssFilename($store);
```
We should strip the protocol from the URL. Currently this won't work if SSL is terminated at a load balancer.

ashsmith replied:
We should fetch the media url properly: `Mage::getBaseUrl(Mage_Core_Model_Store::URL_TYPE_MEDIA);`

If you're using a CDN, currently the URL will only be the base url of the store.

Also, it's possible SSID parameters to be included in this URL.

punkstar replied:
I implemented this like:

```
/**
 * Generate the URL to access the stylesheet for the store.
 *
 * @param null $store
 * @return string
 */
public function getCssFileUrl($store = null)
{
    $url_parts = [
        Mage::getBaseUrl(Mage_Core_Model_Store::URL_TYPE_MEDIA),
        $this->getCssPath($store),
        $this->getCssFilename($store)
    ];

    array_walk($url_parts, function (&$item) {
        $item = trim($item, '/');
    });

    $url = join('/', $url_parts);

    return $this->stripProtocolFromUrl($url);
}
```

ashsmith commented:
Comments came as the result of some internal patches. Just making a note here so I can amend later.

Commit:
Ensure directory is correct, use media url for css file url.

- Fixes issue where the directory path of the generated css file could be missing a directory slash.
- Fixes issue where if a CDN is used, the generated file won't use the CDN URL. Fixes potential issue where SID parameters are included in the URL, and breaks file path.

punkstar commented:
![](https://media.giphy.com/media/uLGs2Enp9RYze/200w_d.gif)
